### PR TITLE
Ensure that quickstart examples actually work.

### LINKED ISSUE #4394: PubSub: Quickstart examples for Pub/Sub subscriber not working
[Subscription quickstart documentation](https://github.com/GoogleCloudPlatform/google-cloud-python/tree/master/pubsub) for Pub/Sub library is not correct in a couple of points. 

1.  The final `subscription` here should be `subscription_name`.
    ```python
    subscription = subscriber.create_subscription(topic, subscription)
    ```
    Using it as is throws the following error: `NameError: name 'subscription' is not defined`

2.  After modifying the line from 1 above to
    ```python
    subscription = subscriber.create_subscription(topic, subscription_name)
    ```
    it throws another error:
    ```python
    Traceback (most recent call last):
      File "receive.py", line 19, in <module>
        subscription = subscriber.create_subscription(topic, subscription_name)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/cloud/pubsub_v1/_gapic.py", line 42, in <lambda>
        fx = lambda self, *a, **kw: wrapped_fx(self.api, *a, **kw)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/cloud/gapic/pubsub/v1/subscriber_client.py", line 445, in create_subscription
        return self._create_subscription(request, options)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/gax/api_callable.py", line 452, in inner
        return api_caller(api_call, this_settings, request)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/gax/api_callable.py", line 438, in base_caller
        return api_call(*args)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/gax/api_callable.py", line 376, in inner
        return a_func(*args, **kwargs)
      File "/path-to-file/pubsub/env/local/lib/python2.7/site-packages/google/gax/retry.py", line 127, in inner
        ' classified as transient', exception)
    google.gax.errors.RetryError: RetryError(Exception occurred in retry method that was not classified as transient, caused by <_Rendezvous of RPC that terminated with (StatusCode.INVALID_ARGUMENT, Invalid resource name given (name=projects/PROJECTID/topics/TOPICNAME). Refer to https://cloud.google.com/pubsub/overview#names for more information.)>)
    ```
    Note that I've copied/pasted the code example and edited the projectID, topic name and subscription name to be valid for my project. Also note that using the example from [readthedocs.io](http://google-cloud-python.readthedocs.io/en/latest/pubsub/subscriber/index.html) (which is the same) generates the exact same error

### PR DESCRIPTION
**Pull request by tseaver**
Closes #4394.

### CODE REVIEW

File: pubsub/README.rst [hunk 1]
```diff
@@ -90,15 +90,16 @@ the topic, and subscribe to that.
...
         project_id=os.getenv('GOOGLE_CLOUD_PROJECT'),
         topic='MY_TOPIC_NAME',  # Set this to something appropriate.
     )
     subscription_name = 'projects/{project_id}/subscriptions/{sub}'.format(
         project_id=os.getenv('GOOGLE_CLOUD_PROJECT'),
         sub='MY_SUBSCRIPTION_NAME',  # Set this to something appropriate.
     )
-    subscription = subscriber.create_subscription(topic, subscription)
+    subscription = subscriber.create_subscription(
+        name=subscription_name, topic=topic_name)
```
Comment (dhermes) [inline pubsub/README.rst]: My preference would be to leave the names out unless these are kwargs in the method itself, though I know you feel differently @tseaver

**Comment by tseaver** [comment]
@dhermes adding the names makes the call site more robust:  back-end / generated code which changes the ordering won't break us, and when they *do* change the name, we get a nice, informative `TypeError`, instead of staring at the obscure back-end message:

```python
(StatusCode.INVALID_ARGUMENT, Invalid resource name given (name=projects/PROJECTID/topics/TOPICNAME). Refer to https://cloud.google.com/pubsub/overview#names for more information.)
```

Since that string *is* a valid topic resource name, I stared at that for a long whilve before figuring out that the error was that `topic_name` and `name` were reversed.

**Comment by duggelz** [comment]
Does anyone have any experience with running code/tests inside documentation via sphinx?  E.g http://www.sphinx-doc.org/en/stable/ext/doctest.html

**Comment by dhermes** [comment]
I have lots of it! Hit me up on Hangouts?

**Pull request merged**